test/clj/chronograph/handlers/task_test.clj

```diff
@@ -0,0 +1,104 @@
+            request {:params {:organization-id random-org-id}}
+            response (task/index request)]
+        (is (= 404 (:status response)))))))
+
+(deftest update-test
+  (let [user (factories/create-user)
+        {organization-id :organizations/id} (factories/create-organization (:users/id user))
+        {task-id :tasks/id} (factories/create-task organization-id)
+        now (time/now)]
+    (testing "Returns 404 if the organization is not found"
+      (let [random-org-id (rand-int 10) ;; TODO: Change this random value
+            request {:params {:organization-id random-org-id}}
+            response (task/update request)]
+        (is (= 404 (:status response)))))
+    (testing "Returns 404 if the task is not found"
+      (let [random-task-id (rand-int 10)
+            request {:params {:organization-id organization-id
+                              :task-id random-task-id}}
+            response (task/update request)]
+        (is (= 404 (:status response)))))
+    (testing "It returns 200 and the updated task if updates are valid"
+      ;; TODO: generate the updates
+      (with-redefs [time/now (constantly now)]
+        (let [updates {:name "Updated value" :description "Updated description"}
+              request {:params {:organization-id organization-id
+                                :task-id task-id}
+                       :body {:updates updates}}
+              response (task/update request)
+              {:tasks/keys [name description updated-at]} (:body response)]
+          (is (= 200 (:status response)))
+          (is (= updates {:name name
+                          :description description}))
+          (is (= now updated-at)))))))
+
+(deftest archive-test
+  (let [user (factories/create-user)
+        {organization-id :organizations/id} (factories/create-organization (:users/id user))
+        {task-id :tasks/id} (factories/create-task organization-id)
+        now (time/now)]
+    (testing "Returns 404 if the organization is not found"
+      (let [random-org-id (rand-int 10) ;; TODO: Change this random value
+            request {:params {:organization-id random-org-id}}
+            response (task/archive request)]
+        (is (= 404 (:status response)))))
+    (testing "Returns 404 if the task is not found"
+      (let [random-task-id (rand-int 10)
+            request {:params {:organization-id organization-id
+                              :task-id random-task-id}}
+            response (task/archive request)]
+        (is (= 404 (:status response)))))
+    (testing "It returns 200 and archives the tasks"
+      (with-redefs [time/now (constantly now)]
+        (let [request {:params {:organization-id organization-id
+                                :task-id task-id} }
+              response (task/archive request)
+              {:tasks/keys [archived-at updated-at]} (:body response)]
+          (is (= 200 (:status response)))
+          (is (= now archived-at))
```

Comment:
Alternative idea: Instead of freezing time, I opted to check that the new value of updated-at / archived-at is greater than the old value. e.g.  https://github.com/nilenso/chronograph/pull/16/files#diff-37e3a3d1de62325e701ee421795fed98R208